src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3544,53 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL)
+                {
+                    uint8_t scanChannel;
+                    uint16_t scanDuration;
+
+                    parsedLength = spinel_datatype_unpack(
```

Comment:
Admittedly, I am surprised that you find my original suggestion as being less convenient than receiving an asynchronous update.